server/utils/utils.go

```diff
@@ -54,3 +55,59 @@ func StructToMap(v interface{}) (m map[string]interface{}) {
 	_ = json.Unmarshal(b, &m)
 	return
 }
+
+// GenerateBlockIDs generates new IDs for all the blocks of the list,
+// keeping consistent any references that other blocks would made to
+// the original IDs, so a tree of blocks can get new IDs and maintain
+// its shape.
+func GenerateBlockIDs(blocks []model.Block) []model.Block {
+	blockIDs := map[string]bool{}
+	referenceIDs := map[string]bool{}
+	for _, block := range blocks {
+		if _, ok := blockIDs[block.ID]; !ok {
+			blockIDs[block.ID] = true
+		}
+
+		if _, ok := referenceIDs[block.RootID]; !ok {
+			referenceIDs[block.RootID] = true
+		}
+		if _, ok := referenceIDs[block.ParentID]; !ok {
+			referenceIDs[block.ParentID] = true
+		}
+	}
+
+	newIDs := map[string]string{}
+	for id := range blockIDs {
+		for referenceID := range referenceIDs {
+			if id == referenceID {
+				newIDs[id] = NewID(IDTypeBlock)
+				continue
+			}
+		}
+	}
+
+	getExistingOrOldID := func(id string) string {
+		if existingID, ok := newIDs[id]; ok {
+			return existingID
+		}
+		return id
+	}
+
+	getExistingOrNewID := func(id string) string {
+		if existingID, ok := newIDs[id]; ok {
+			return existingID
+		}
+		return NewID(IDTypeBlock)
```

Comment:
I created a PR with a method to convert block types to IDType.  https://github.com/mattermost/focalboard/pull/1715   That can be used to pass the correct IDType here.

Reply:
That PR was revised here https://github.com/mattermost/focalboard/pull/1744 to move the new method to a better place.

Reply:
Updated on 71cc8f7, PTAL